river/tree/nodes/last_nodes.py

```diff
@@ -0,0 +1,187 @@
+from __future__ import annotations
+
+from ..utils import do_naive_bayes_prediction
+from .htc_nodes import LeafMajorityClass
+
+
+class LeafMajorityClassWithDetector(LeafMajorityClass):
+    """Leaf that always predicts the majority class.
+
+    Parameters
+    ----------
+    stats
+        Initial class observations.
+    depth
+        The depth of the node.
+    splitter
+        The numeric attribute observer algorithm used to monitor target statistics
+        and perform split attempts.
+    kwargs
+        Other parameters passed to the learning node.
+    """
+
+    def __init__(self, stats, depth, splitter, change_detector, split_criterion=None, **kwargs):
+        super().__init__(stats, depth, splitter, **kwargs)
+        self.change_detector = change_detector
+        self.split_criterion = (
+            split_criterion  # if None, the change detector will have binary inputs
+        )
```

Comment:
*Update:* trees actually don't have a split criterion object. `tree.split_criterion` is currently a string with the alias of a valid split criterion. In the future, we might refactor a bit and make this property become the criteria directly. This however would imply a significant deal of refactoring and documentation as the users would need to pass an object rather than a string when instantiating the tree model.

For now, let's keep things as they are to avoid too many changes outside the scope of this PR.